File: src/com/aopphp/go/GoAopFileType.java
package com.aopphp.go;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class GoAopFileType extends LanguageFileType {
    public static final GoAopFileType INSTANCE = new GoAopFileType();

    private GoAopFileType() {
        super(GoAopLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Go! AOP Pointcut";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Go! AOP Pointcut Expression Syntax";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return ".goaop";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return GoAopIcons.FILE;
    }
}

File: src/main/kotlin/com/aopphp/go/GoAopFileType.kt
package com.aopphp.go

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

object GoAopFileType : LanguageFileType(PointcutQueryLanguage) {
    override fun getName() = "Go! AOP Pointcut"
    override fun getDescription() = "Go! AOP Pointcut Expression Syntax"
    override fun getDefaultExtension() = "goaop"
    override fun getIcon(): Icon = GoAopIcons.FILE
}
